environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentService.java

```diff
@@ -447,4 +452,18 @@ public EnumSet<Crn.ResourceType> getSupportedCrnResourceTypes() {
     public Optional<Environment> getById(Long environmentId) {
         return environmentRepository.findById(environmentId);
     }
+
+    public Environment updateTunnelByEnvironmentId(Long environmentId, Tunnel tunnel) {
+        Optional<Environment> optEnv = getById(environmentId);
+        if (optEnv.isPresent()) {
+            Environment env = optEnv.get();
+            ExperimentalFeatures experimentalFeatures = env.getExperimentalFeaturesJson();
+            experimentalFeatures.setTunnel(tunnel);
```

Comment:
Can we have log about the update of tunnel type from what to what?

Reply:
ok